### PR TITLE
More fix me sprites to help find more broken stuff, some broken sprite fixes

### PR DESCRIPTION
**Pull request by Skeletonman0**
[BUG] [OBJECTS] [SPRITES]
## About the PR 
This pr adds fix me sprites for humans (not for mutant races in this PR), borgs, clothes, doors, vending machines, storage/lockers/crates, and pods/subs

this exposed some otherwise invisible issues, such as borg limbs putting an invalid icon state ontop of your sprite
and belt/id slot items doing the same.

image

those two issues should mostly be fixed in this PR. other than those, i did not notice any other obvious "FIX ME" signs immediately as a result of this.

to fix the belt/id slots creating broken sprites, i made it use the wear_state var instead of icon_state and item_state.
to fix the borg issue, i found that it was generating an image using a invalid icon state put together using a few vars.
i changed it slightly to stop this.

## Why's this needed? 
Fix me sprites help find bugs that cause invisible things which would otherwise go unnoticed

Everyone spawning in with a invalid icon state from their PDA in their belt or id slot seemed like a bad thing

an example of the PDA thing:

https://gyazo.com/2b45a6f8245da1a8dcb3205ebf7d8351

### CODE REVIEW

**Comment by frawhst** [comment]
!merge_upstream

File: code/mob/living/carbon/human/procs/update_icon.dm [hunk 1]
```diff
@@ -118,9 +118,9 @@
 		UpdateOverlays(null, "material_suit")
 
 	// ID
-	if (src.wear_id)
+	if (src.wear_id && src.wear_id?.wear_state)
```
Comment (frawhst) [inline code/mob/living/carbon/human/procs/update_icon.dm:121]:
```suggestion
	if (src.wear_id?.wear_state)
```
redundant nullcheck

File: code/mob/living/carbon/human/procs/update_icon.dm [hunk 2]
```diff
@@ -118,9 +118,9 @@
...
 		wear_sanity_check(src.wear_id)
-		var/wear_state = src.wear_id.wear_state || src.wear_id.icon_state
+		var/wear_state = src.wear_id.wear_state// || src.wear_id.icon_state
```
Comment (frawhst) [inline code/mob/living/carbon/human/procs/update_icon.dm:123]:
```suggestion
		var/wear_state = src.wear_id.wear_state
```
just delete code if you don't need it anymore

File: code/mob/living/carbon/human/procs/update_icon.dm [hunk 3]
```diff
@@ -560,9 +560,9 @@
 		UpdateOverlays(null, "wear_head_blood")
 		UpdateOverlays(null, "material_head")
 	// Belt
-	if (src.belt)
+	if (src.belt && src.belt?.wear_state)
```
Comment (frawhst) [inline code/mob/living/carbon/human/procs/update_icon.dm:563]:
```suggestion
	if (src.belt?.wear_state)
```
see above

File: code/mob/living/carbon/human/procs/update_icon.dm [hunk 4]
```diff
@@ -560,9 +560,9 @@
...
 		wear_sanity_check(src.belt)
-		var/wear_state = src.belt.wear_state || src.belt.item_state || src.belt.icon_state
+		var/wear_state = src.belt.wear_state// || src.belt.item_state || src.belt.icon_state
```
Comment (frawhst) [inline code/mob/living/carbon/human/procs/update_icon.dm:565]:
```suggestion
		var/wear_state = src.belt.wear_state
```
.

File: code/obj/item.dm [hunk 5]
```diff
@@ -286,8 +286,15 @@
...
+
+	// this is here so things wont be dumb
+	var/wstate = icon_state
+	if (wear_state)
+		wstate = wear_state
+
 	wear_image = image(wear_image_icon)
-	wear_image.icon_state = icon_state //Why was this null until someone actually wore it? Made manipulation impossible.
+	wear_image.icon_state = wstate //Why was this null until someone actually wore it? Made manipulation impossible.
+
```
Comment (frawhst) [inline code/obj/item.dm:296]:
```suggestion
	wear_image.icon_state = icon_state || wear_state
```
This makes it so the `wear_image` uses `icon_state` if it's set and `wear_state` if it isn't, same as above but in way less lines.

File: code/obj/item.dm [hunk 6]
```diff
@@ -286,8 +286,15 @@
 
 /obj/item/New()
 	// this is dumb but it won't let me initialize vars to image() for some reason
+
+	// this is here so things wont be dumb
+	var/wstate = icon_state
+	if (wear_state)
+		wstate = wear_state
+
```
Comment (frawhst) [inline code/obj/item.dm:293]:
```suggestion
```
See above for why this isn't needed

File: code/obj/item/card.dm [hunk 7]
```diff
@@ -106,6 +106,7 @@ GAUNTLET CARDS
 
 /obj/item/card/id/New()
 	..()
+	src.wear_state = icon_state // fix for missing wear states
```
Comment (frawhst) [inline code/obj/item/card.dm:109]: Isn't this set in `item/new()`?

File: code/obj/item/device/pda2/pda2.dm [hunk 8]
```diff
@@ -5,7 +5,7 @@
 	desc = "A portable microcomputer by Thinktronic Systems, LTD. It has a slot for an ID card, and a hole to put a pen into."
 	icon = 'icons/obj/items/pda.dmi'
 	icon_state = "pda"
-	item_state = "pda"
+	//item_state = "pda" NO, STOP FIX ME SPRITING
```
Comment (frawhst) [inline code/obj/item/device/pda2/pda2.dm:8]:
```suggestion
```
idk what the comment means but I'm assuming this is unnecessary or broken and thus Delete

**Review by frawhst — COMMENTED** [review]
Good concept, should find Bugz

File: code/obj/item/device/pda2/pda2.dm (hunk 8 above)
Comment (Skeletonman0) [reply code/obj/item/device/pda2/pda2.dm]: whoops, it actually does need that. the reason it was commented out was that it was trying to use the inhand iconstate due to belt overlay code using it previously

File: code/obj/item/card.dm (hunk 7 above)
Comment (Skeletonman0) [reply code/obj/item/card.dm]: it should be, yeah. removed that.

**Comment by frawhst** [comment]
!merge_upstream

**Review by frawhst — APPROVED** [review]

**Pull request merged**